### PR TITLE
Add set quiet time

### PR DESCRIPTION
**Pull request by eosterberg**
Also, add an "are_notifications_enabled" method for iOS

### CODE REVIEW

File: ios/ReactNativeUAIOS/ReactNativeUAIOS.m [hunk 1]
```diff
@@ -146,6 +146,23 @@ - (void)dispatchEvent:(NSString *)event body:(NSDictionary *)notification {
...
+    [[UAirship push] setQuietTimeStartHour:[time[@"startHour"] unsignedIntegerValue] startMinute:[time[@"startMinute"] unsignedIntegerValue] endHour:[time[@"endHour"] unsignedIntegerValue] endMinute:[time[@"endMinute"] unsignedIntegerValue]];
+    [[UAirship push] updateRegistration];
+}
+
+RCT_EXPORT_METHOD(setQuietTimeEnabled:(nonnull NSNumber *)enabled) {
+    [[UAirship push] setQuietTimeEnabled:[enabled boolValue]];
+    [[UAirship push] updateRegistration];
+}
+
+RCT_EXPORT_METHOD(areNotificationsEnabled:(RCTResponseSenderBlock)callback)
```
Comment (rlepinski) [inline ios/ReactNativeUAIOS/ReactNativeUAIOS.m]: there is a new line here compared to the other methods. `callback) {`

File: ios/ReactNativeUAIOS/ReactNativeUAIOS.m [hunk 2]
```diff
@@ -146,6 +146,23 @@ - (void)dispatchEvent:(NSString *)event body:(NSDictionary *)notification {
...
+{
+    UIUserNotificationSettings *grantedSettings = [[UIApplication sharedApplication] currentUserNotificationSettings];
```
Comment (rlepinski) [inline ios/ReactNativeUAIOS/ReactNativeUAIOS.m]: This is using deprecated iOS 8 & 9 apis. Instead use:
```
BOOL enabled = [UAirship push].authorizedNotificationOptions != 0;
```

**Comment by rlepinski** [comment]
Android you can determine if notifications are opted in with `UAirship.shared().getPushManager().isOptIn()`

File: android/src/main/java/com/globo/reactnativeua/ReactNativeUA.java [hunk 3]
```diff
@@ -49,6 +54,25 @@ public void removeTag(String tag) {
...
+        UAirship.shared().getPushManager().setQuietTimeInterval(startDate, endDate);
+
+        } catch (ParseException ex) {
+          ex.printStackTrace();
+        }
+    }
+
+    @ReactMethod
+    public void setQuietTimeEnabled(Boolean enabled) {
+      UAirship.shared().getPushManager().setQuietTimeEnabled(enabled);
```
Comment (rlepinski) [inline android/src/main/java/com/globo/reactnativeua/ReactNativeUA.java]: whitespacing is messed up here. Need 4 spaces `     UAirship.shared()`

File: android/src/main/java/com/globo/reactnativeua/ReactNativeUA.java [hunk 4]
```diff
@@ -49,6 +54,25 @@ public void removeTag(String tag) {
     }
 
     @ReactMethod
+    public void setQuietTime(ReadableMap time) {
+      try {
```
Comment (rlepinski) [inline android/src/main/java/com/globo/reactnativeua/ReactNativeUA.java]: More white spacing issues in this method. 4 spaces for indents. Everything in the try needs another 4 spaces.

File: android/src/main/java/com/globo/reactnativeua/ReactNativeUA.java [hunk 5]
```diff
@@ -49,6 +54,25 @@ public void removeTag(String tag) {
...
+        SimpleDateFormat formatter = new SimpleDateFormat("hh:mm", Locale.getDefault());
+        Date startDate = formatter.parse(time.getInt("startHour") + ":" + time.getInt("startMinute"));
+        Date endDate = formatter.parse(time.getInt("endHour") + ":" + time.getInt("endMinute"));
+
+        UAirship.shared().getPushManager().setQuietTimeInterval(startDate, endDate);
+
+        } catch (ParseException ex) {
+          ex.printStackTrace();
```
Comment (rlepinski) [inline android/src/main/java/com/globo/reactnativeua/ReactNativeUA.java]: You should log this instead of just printing it

**Comment by eosterberg** [comment]
Thanks, working on it...

**Comment by evandroeisinger** [comment]
@emilianoeloi @guilhermebruzzi could you validade this PR when @eosterberg finish his improvement?

Thanks @eosterberg and @rlepinski (as always 😄 ) for the contribution, soon as possible we will release a new version with you improvements.

**Comment by rlepinski** [comment]
Looks good to me

**Comment by evandroeisinger** [comment]
@eosterberg merged!

**Comment by evandroeisinger** [comment]
@eosterberg could you update the README.md with your improvements? Then we will submit a new version to npm.

**Comment by eosterberg** [comment]
Sure, I´ll update it in a few days.

**Pull request merged**